### PR TITLE
Rename parallel::distributed::SolutionTransfer::prepare_serialization()

### LINKED ISSUE #7566: Rename SolutionTransfer::prepare_serialization
This class has a function `SolutionTransfer::prepare_for_coarsening_and_refinement()`, but the `SolutionTransfer::prepare_serialization()` doesn't have the word "for" in the name. Rename the latter function to `prepare_for_serialization()` to make things consistent.

This came up in the discussion of https://github.com/dealii/dealii/pull/7537#pullrequestreview-188121644

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #7566.

### CODE REVIEW

File: source/distributed/solution_transfer.cc [hunk 1]
```diff
@@ -109,13 +109,34 @@ namespace parallel
...
+    }
+
+
+
+    template <int dim, typename VectorType, typename DoFHandlerType>
+    void
+    SolutionTransfer<dim, VectorType, DoFHandlerType>::
+      prepare_for_serialization(const std::vector<const VectorType *> &all_in)
+    {
+      prepare_for_coarsening_and_refinement(all_in);
```
Comment (davydden) [inline source/distributed/solution_transfer.cc]: prepare_for_serialization ?

**Comment by masterleinad** [comment]
Again, all checks are passing but the status of `ready` is not shown correctly.

**Comment by tjhei** [comment]
> Again, all checks are passing but the status of `ready` is not shown correctly.

I will fix that problem soon.

**Comment by tjhei** [comment]
You copied (the short) implementation. I think it would more sense if the deprecated function calls the new function instead of the copy.

**Pull request merged**